satori/resources/resource_datastore_common.go

```diff
@@ -0,0 +1,306 @@
+	ExcludedQueryPatterns       = "excluded_query_patterns"
+	Identity                    = "identity"
+	IdentityType                = "identity_type"
+)
+var TreatAsMap = map[string]bool{
+	Exclusions:                  true,
+	UnsupportedQueriesCategory:  true,
+	UnassociatedQueriesCategory: true,
+	BaselineSecurityPolicy:      true,
+}
+
+func getDataStoreDefinitionSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		Id: &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "DataStore resource id.",
+		},
+		ParentId: &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Parent resource id.",
+		},
+		Name: &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "DataStore name.",
+		}, Hostname: &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Data provider's FQDN hostname.", // example: snowflakecomputing.com, xyz.redshift.amazonaws.com:5439/dev
+		}, OriginPort: &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Port number description.",
+		}, DataAccessControllerId: &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Host FQDN name.",
+		},
+		CustomIngressPort: &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Port number description.",
+		}, IdentityProviderId: &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Port number description.",
+		}, ProjectIds: &schema.Schema{
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "ProjectIds list of project IDs",
+			Elem:        &schema.Schema{Type: schema.TypeString},
+		}, Type: &schema.Schema{
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "IDs of Satori users that will be set as DataStore owners.",
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		}, IdentityProviderId: &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "IDs of Satori users that will be set as DataStore owners.",
+		},
+		BaselineSecurityPolicy: {
+			Type:        schema.TypeList,
+			Required:    true,
+			MaxItems:    1,
+			Description: "Baseline security policy.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					Type: &schema.Schema{
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "DataStore basepolicy .",
+						Default:     "BASELINE_POLICY",
+					},
+					UnassociatedQueriesCategory: {
+						Type:        schema.TypeList,
+						Required:    true,
+						MaxItems:    1,
+						Description: "UnassociatedQueriesCategory",
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								QueryAction: &schema.Schema{
+									Type:        schema.TypeString, //Default:     "PASS",
```

Comment:
Can we add a description of possible values? So you will be able to remove it from the example....